From 7a19a9f4adbb5e6726a07710c5f21db63a75a677 Mon Sep 17 00:00:00 2001
From: Patrik Flykt <patrik.flykt@intel.com>
Date: Fri, 1 Jul 2022 16:15:57 +0300
Subject: [PATCH 3/3] configure: Check if pkg-config is installed before use

Fix library configuration to work even if pkg-config is not installed.
Check for pkg-config, and wrap its potential use inside an AS_IF() as
the PKGCONFIG variable might not be defined as a result.

Signed-off-by: Patrik Flykt <patrik.flykt@intel.com>
---
 configure.ac | 42 +++++++++++++++++++++++++++++++-----------
 1 file changed, 31 insertions(+), 11 deletions(-)

diff --git a/configure.ac b/configure.ac
index 844ec8f..3f4e1f8 100644
--- a/configure.ac
+++ b/configure.ac
@@ -76,17 +76,31 @@ AC_COMPILE_IFELSE([AC_LANG_PROGRAM([])],
 CFLAGS="$saved_cflags"
 AC_SUBST(NO_UNUSED_CMDLINE_ARG_CFLAGS)
 
-# Check location of systemd unit files
-PKG_PROG_PKG_CONFIG
+# Check for pkgconfig
+AC_MSG_CHECKING([for pkg-config])
+AS_IF([test "x${PKGCONFIG}" = "x"],
+    [
+     AC_MSG_RESULT(no)
+     have_pkgconfig="no"
+    ],
+    [
+     AC_MSG_RESULT(yes)
+     have_pkgconfig="yes"
+    ]
+)
 
 # Check for systemd.
-AS_IF([test "x$PKGCONFIG" = "x"],
-    [AC_MSG_ERROR(pkg-config not found.)],
-    [AC_MSG_CHECKING(for systemd)
+AS_IF([test "x${have_pkgconfig}" = "xyes"],
+    [
+     AC_MSG_CHECKING(for systemd pkg-config support)
      AS_IF([$PKGCONFIG --exists systemd],
-        [AC_MSG_RESULT(yes); have_systemd="yes"],
-        [AC_MSG_RESULT(no);  have_systemd="no"]
-    )]
+        [
+	 AC_MSG_RESULT(yes); have_systemd="yes"
+	],
+        [
+	 AC_MSG_RESULT(no);  have_systemd="no"
+	])
+    ], []
 )
 
 AC_ARG_ENABLE(systemd,
@@ -100,12 +114,18 @@ else
     AC_MSG_CHECKING(whether to use systemd)
     AC_MSG_RESULT(${enable_systemd})
 fi
-AM_CONDITIONAL(HAVE_SYSTEMD, test "${enable_systemd}" != "no")
 
+# Check location of systemd unit files
 AC_ARG_WITH([systemdsystemunitdir],
         AS_HELP_STRING([--with-systemdsystemunitdir=DIR], [Directory for systemd service files]),
-        [], [with_systemdsystemunitdir=$($PKG_CONFIG --variable=systemdsystemunitdir systemd)])
-AC_SUBST([systemdsystemunitdir], [$with_systemdsystemunitdir])
+        [],
+       [AS_IF([test "${have_pkgconfig}" = "yes"],
+            [with_systemdsystemunitdir=$($PKGCONFIG --variable=systemdsystemunitdir systemd)
+             AC_SUBST([systemdsystemunitdir], [$with_systemdsystemunitdir])],
+            [])
+        ])
+
+AM_CONDITIONAL(HAVE_SYSTEMD, test "${enable_systemd}" != "no")
 
 AS_IF([test "${enable_systemd}" != "no"],
 	    [AC_MSG_CHECKING(systemd unit directory); AC_MSG_RESULT(${systemdsystemunitdir})],
-- 
2.36.1

